src/sc-main.c

```diff
@@ -80,8 +81,21 @@ int sc_main(int argc, char **argv)
 #endif				// ifdef HAVE_SECCOMP
 
 	if (geteuid() == 0) {
-		sc_unshare_mount_ns();
-		sc_populate_mount_ns(security_tag);
+		const char *group_name = getenv("SNAP_NAME");
+		if (group_name == NULL) {
+			die("SNAP_NAME is not set");
+		}
+		sc_initialize_ns_groups();
+		struct sc_ns_group *group = NULL;
+		group = sc_open_ns_group(group_name);
+		sc_lock_ns_mutex(group);
+		sc_create_or_join_ns_group(group);
+		if (sc_should_populate_ns_group(group)) {
+			sc_populate_mount_ns(security_tag);
+			sc_preserve_populated_ns_group(group);
+		}
+		sc_unlock_ns_mutex(group);
+		sc_close_ns_group(group);
```

Comment:
This logic seems fine but seems like a lot to have in sc-main.c. Before we had two simple calls to unshare and to populate, but now we have tricky logic and ordering. I think this is fine to land as is, but perhaps we could abstract this out to a wrapper function in ns-support.c and then sc-main.c can just call that. This would also provide an opportunity to document in an easier to understand fashion by documenting the high-level wrapper function and what it does, and then leave documenting the nitty-gritty details of what it uses to above the respective functions.